common/controllers/form-wizard.js

```diff
@@ -1,10 +1,51 @@
 const Sentry = require('@sentry/node')
 const { Controller } = require('hmpo-form-wizard')
-const { map, fromPairs } = require('lodash')
+const { flatten, map, fromPairs, forEach } = require('lodash')
 
 const fieldHelpers = require('../helpers/field')
 
 class FormController extends Controller {
+  middlewareSetup() {
+    super.middlewareSetup()
+    this.use(this.setupConditionalFields)
+  }
+
+  setupConditionalFields(req, res, next) {
+    const allFields = req.form.options.allFields
+    const stepFields = req.form.options.fields
+    const dependentFields = {}
+
+    forEach(stepFields, (field, key) => {
+      if (field.items) {
+        field.items.forEach(item => {
```

Comment:
The code boils down to:

```js
const dependentFields = {} // accumulator
conditionalFields.forEach(it => {
  /* code */
  /* add to accumulator /*
})
```

This is a hand-rolled `.reduce`. You could replace the `forEach` with `reduce` (or `fold`, not sure what's called in lodash).

Reply:
The slight issue with that is that the `dependentFields` needs to be built outside of the parent loop of `conditionalFields`. I'll see if I can tweak the solution to use `reduce` though

Reply:
@danielepolencic I wasn't really able to get it neatly working so have tweaked it to use early return but kept the `forEach` for now.

If you can post a code snippet I'd be happy to push another change to fix it after?

Reply:
No worries, I had a go at it and I understand your point better now. It's not straightforward.
The code has 3 nested layers of loops. It could be refactored and flattened like this:

```js
  _(stepFields)
    .chain()
    .filter(it => Array.isArray(it.items))
    .map((item, fieldName) => {
      return _(flatten([item.conditional || []]))
        .chain()
        .map(it => allFields[it])
        .filter(it => !!it)
        .map(it => {
          return {
            ...it,
            // tell the form wizard to not output field initially as it will be nested
            skip: true,
            // set dependent object for validation
            dependent: {
              field: fieldName,
              value: item.value,
            },
          }
        })
        .value()
    })
    .value()
```

I'm not terribly familiar with lodash, so the above might need tweaking (also, no idea on how to chain functions, I usually use the native `.map`, `.reduce`, etc.).

You don't have to implement the change and I don't see a problem if it's addressed later.